x-pack/plugins/maps/public/shared/layers/sources/es_search_source/es_search_source.js

```diff
@@ -136,7 +202,26 @@ export class ESSearchSource extends AbstractESSource {
     // By setting "fields", SearchSource removes all of defaults
     searchSource.setField('fields', searchFilters.fieldNames);
 
-    let featureCollection;
+    const resp = await this._runEsQuery(layerName, searchSource, 'Elasticsearch document request');
+
+    return {
+      hits: resp.hits.hits,
+      meta: {
+        areResultsTrimmed: resp.hits.total > resp.hits.hits.length
```

Comment:
In case `_runEsQuery` throws, we might want to use `_.get(resp, ....)` with defaults for the values here as well

Reply:
If _runEsQuery throws then it won't matter because our function will throw since _runEsQuery is not wrapped in a try/catch. This is the expected behavior so vector_layer can catch the exception and call onLoadError